server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4554,6 +4581,112 @@ public void syncSegmentsFromRemoteSegmentStore(boolean overrideLocal, boolean re
         }
     }
 
+    private RemoteSegmentStoreDirectory getRemoteSegmentDirectoryForShard() {
+        FilterDirectory remoteStoreDirectory = (FilterDirectory) remoteStore.directory();
+        assert remoteStoreDirectory.getDelegate() instanceof FilterDirectory
+            : "Store.directory is not enclosing an instance of FilterDirectory";
+        FilterDirectory byteSizeCachingStoreDirectory = (FilterDirectory) remoteStoreDirectory.getDelegate();
+        final Directory remoteDirectory = byteSizeCachingStoreDirectory.getDelegate();
+        assert remoteDirectory instanceof RemoteSegmentStoreDirectory : "remoteDirectory is not an instance of RemoteSegmentStoreDirectory";
+        return ((RemoteSegmentStoreDirectory) remoteDirectory);
+    }
+
+    /**
+     * Downloads segments from remote segment store.
+     * @param overrideLocal flag to override local segment files with those in remote store
+     * @throws IOException if exception occurs while reading segments from remote store
+     */
+    public void syncSegmentsFromGivenRemoteSegmentStore(
+        boolean overrideLocal,
+        RemoteSegmentStoreDirectory tempRemoteSegmentDirectory,
+        long primaryTerm,
+        long commitGeneration,
+        boolean syncTillCommitPoint
```

Comment:
do we really  need this or can it be assumed true ?

Reply:
For current usecase, we can assume this true, but just kept it as method parameter, if we need it later.

Reply:
removed it